app/dashboard/router.py

```diff
@@ -264,6 +265,14 @@ def get_queryset(self):
             end = start + int(limit)
             queryset = queryset[start:end]
 
+
+        # save search history 
+        SearchHistory.objects.create(
+            user=self.request.user if not self.request.user.is_anonymous else None,
```

Comment:
for some reason, i get `AnonymousUser` everytime I call `self.request.user` in the router.py context.. it seems to not be Anonymous in regular views

Reply:
@owocki Fixed in:  https://github.com/gitcoinco/web/pull/2756/commits/04bf2fdcd49244dc1f173396d9277830a402bbf3

Reply:
Ref:  [If no class authenticates, request.user will be set to an instance of django.contrib.auth.models.AnonymousUser, and request.auth will be set to None.](https://www.django-rest-framework.org/api-guide/authentication/#how-authentication-is-determined )